lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandler.java

```diff
@@ -266,24 +270,38 @@ private boolean checkRequiresAdditionalEvidence(List<SignedJWT> credentials)
             List<SignedJWT> evidenceCredentials =
                     userIdentityService.filterValidCredentials(credentials);
             if (evidenceCredentials.size() == 1) {
-                var credentialIssuer = evidenceCredentials.get(0).getJWTClaimsSet().getIssuer();
-                boolean requiresAdditionalEvidence =
-                        configService
-                                .getCredentialIssuerActiveConnectionConfig(credentialIssuer)
-                                .isRequiresAdditionalEvidence();
-                LOGGER.info(
-                        new StringMapMessage()
-                                .with(
-                                        LOG_MESSAGE_DESCRIPTION.getFieldName(),
-                                        "GPG45 profile should not not be set for this credential.")
-                                .with("credentialIssuer", credentialIssuer)
-                                .with("requiresAdditionalEvidence", requiresAdditionalEvidence));
-                return requiresAdditionalEvidence;
+                var credentialIssuer =
+                        getCRIWithCredentialIssuer(
+                                evidenceCredentials.get(0).getJWTClaimsSet().getIssuer());
+                CredentialIssuerConfig credentialIssuerConfig =
+                        configService.getCredentialIssuerActiveConnectionConfig(credentialIssuer);
+                if (credentialIssuerConfig != null) {
+                    boolean requiresAdditionalEvidence =
+                            credentialIssuerConfig.isRequiresAdditionalEvidence();
+                    LOGGER.info(
+                            new StringMapMessage()
+                                    .with(
+                                            LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                            "GPG45 profile should not not be set for this credential.")
+                                    .with("credentialIssuer", credentialIssuer)
+                                    .with(
+                                            "requiresAdditionalEvidence",
+                                            requiresAdditionalEvidence));
+                    return requiresAdditionalEvidence;
+                }
             }
         }
         return false;
     }
 
+    public String getCRIWithCredentialIssuer(String iss) {
```

Comment:
I don't think we can do this - the issuer id doesn't necessarily match the ID in the configuration, especially in production. E.g. the `ukPassport` CRI has issuer ID `https://review-p.account.gov.uk`

If you'd like to fix it now, then you'll need to pass around `VcStoreItem`s and refactor the other places that use the credentials accordingly. If you think that's too painful, then you can add your new call to the list on https://govukverify.atlassian.net/browse/PYIC-3997